tiledb/sm/enums/datatype.h

```diff
@@ -52,7 +52,7 @@ enum class Datatype : uint8_t {
 };
 
 /** Returns the datatype size. */
-inline uint64_t datatype_size(Datatype type) {
+inline uint64_t datatype_size(Datatype type) noexcept {
```

Comment:
Should the `bool` functions at the end also be marked `noexcept`?

Reply:
I didn't look at them because they're not exposed through the C API.

This function, though, should actually throw on an invalid argument error rather than mask the error with a zero return value.